Init transaction filter

danielgranhao opened the pull request.

gcomte commented on src/chain_access.rs:
```diff
@@ -0,0 +1,190 @@
+        let builder = Builder::new("http://localhost:9");
+        let esplora_client = Arc::new(builder.build_async().unwrap());
+
+        let filter = LipaChainAccess::new(esplora_client.clone());
```
Create a `fn setup()` within the `tests` mod for the code up until here.
This will duplicated code and particularly prevent us from having to read that port 9 is a discard port over and over again

gcomte commented on src/chain_access.rs:
```diff
@@ -0,0 +1,190 @@
+        let filter = LipaChainAccess::new(esplora_client.clone());
+
+        // Get initial length of filter vecs
+        let initial_queued_tx_amount = filter.queued_txs.lock().unwrap().len();
```
AFAIU, these lengths will always be `0`, which is what the previous test proves.

So you might as well just write:
`let initial_queued_tx_amount = 0;`

or even remove it altogether and do it in the `asser_eq!` instead:
```
assert_eq!(filter.queued_txs.lock().unwrap().len(), 1);
assert_eq!(filter.queued_outputs.lock().unwrap().len(), 0);
assert_eq!(filter.watched_txs.lock().unwrap().len(), 0);
assert_eq!filter.watched_outputs.lock().unwrap().len(), 0);
```

Otherwise it just feels very verbose to me, or I'm missing something

danielgranhao commented on src/lib.rs:
```diff
@@ -79,6 +81,9 @@ impl LightningNode {
             warn!("Object storage is unhealty");
         }
 
+        // Initialize the Transaction Filter
```
The step numbering is currently being done according to this [tutorial](https://lightningdevkit.org/tutorials/build_a_node_in_rust/). In there, as initialising a filter is optional, they don't number it. If we number this step, the following steps will be correctly numbered when compared to the tutorial.

What do you mean about it being more than initialising?

andrei-21 replied:
If we follow the tutorial, then it should be after step 4.
https://lightningdevkit.org/tutorials/build_a_node_in_rust/#optional-initialize-the-transaction-filter.

gcomte replied:
I don't like because it feels like being part of the step `Initialize Persist` which it isn't, as a matter of fact nothing is being persisted here. So I think it is confusing. Even calling it `Step X` could clear that up. Or `Step NaN`. I agree it's tricky if you find it valuable to stick to the tutorial steps.

danielgranhao commented:
Ok, I think I addressed your concerns. I've just rebased from develop so that this already provides the `ChainMonitor` with the `Filter` implementation.

andrei-21 commented on src/lib.rs:
```diff
@@ -94,9 +96,12 @@ impl LightningNode {
         // Step 5. Initialize the ChainMonitor
         let chain_monitor = Arc::new(ChainMonitor::new(
-            None,
+            Some(filter.clone()),
```
Should we follow the practice to clone `Arc` with `Arc::clone()` (see examples below)?
